### PR TITLE
Add Private-Token to redacted headers

### PR DESCRIPTION
**Pull request by pkowalcze**
Closes https://github.com/pitgull/pitgull/issues/124

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,4 @@ target/
 .vscode/
 **/*.worksheet.sc
 docker/pass.sh
+**/.bsp/**
```
Comment (kubukoz) [inline .gitignore:11]: I think we can just do `**/.bsp`

File: src/main/scala/io/pg/Main.scala [hunk 2]
```diff
@@ -49,12 +49,18 @@ object Main extends IOApp {
       Sync[F].delay(OdinInterop.globalLogger.set(logger.mapK(Effect.toIOK).some))
     }
 
-  def mkServer[F[_]: Logger: ConcurrentEffect: Timer](http: HttpConfig, meta: MetaConfig, routes: HttpApp[F]) = {
+  def mkServer[F[_]: Logger: ConcurrentEffect: Timer](
+    http: HttpConfig,
```
Comment (kubukoz) [inline src/main/scala/io/pg/Main.scala:53]: At this point I'm wondering if we should just pass the entire config, but I'm too lazy to check how much it'd impact the tests

File: src/main/scala/io/pg/Main.scala [hunk 3]
```diff
@@ -49,12 +49,18 @@ object Main extends IOApp {
...
+    meta: MetaConfig,
+    middlewareConfig: MiddlewareConfig,
+    routes: HttpApp[F]
+  ) = {
     val app = middleware
       .Logger
       .httpApp(
         logHeaders = true,
         logBody = true,
+        redactHeadersWhen = middlewareConfig.sensitiveHeaders.contains,
```
Comment (kubukoz) [inline src/main/scala/io/pg/Main.scala:63]: Is the endpoint actually receiving a token header? The ticket was created with the client middleware in mind

**Review by kubukoz — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (kubukoz) [inline .gitignore:11]:
```suggestion
**/.bsp
```

File: src/main/scala/io/pg/Main.scala (hunk 2 above)
Comment (majk-p) [reply src/main/scala/io/pg/Main.scala:53]: Looks like we are already passing most of `AppConfig` already:
```scala
  val appConfig: ConfigValue[AppConfig] =
    (httpConfig, metaConfig, gitConfig, queuesConfig, middlewareConfig).parMapN(apply)
```
Switching to full config would require providing fake `gitConfig` and `queuesConfig` in tests. 

@pkowalcze wdyt?

**Review by majk-p — APPROVED** [review]

File: src/main/scala/io/pg/Main.scala (hunk 2 above)
Comment (pkowalcze) [reply src/main/scala/io/pg/Main.scala]: Those fake instances are actually already provided in tests, so the only impact is a shorter argument list here - so I passed the entire config here

File: src/main/scala/io/pg/Main.scala (hunk 3 above)
Comment (pkowalcze) [reply src/main/scala/io/pg/Main.scala]: It doesn't, I've been a little overzealous with this change, removed

**Comment by kubukoz** [comment]
LBGTM

**Pull request merged**